dlib/image_transforms/interpolation.h

```diff
@@ -310,7 +310,9 @@ namespace dlib
                     pixel_to_vector<double>(img[r+1][c  ])(i),
                     pixel_to_vector<double>(img[r+1][c+1])(i));
             typename image_view_type::pixel_type temp;
-            vector_to_pixel(temp, pvout);
+            const auto min_val = pixel_traits<typename image_view_type::pixel_type>::min();
+            const auto max_val = pixel_traits<typename image_view_type::pixel_type>::max();
+            vector_to_pixel(temp, clamp(pvout, min_val, max_val));
             assign_pixel(result, temp);
```

Comment:
You can use the alias `pixel_type_t<image_view_type>` if you want.
So:
* `const auto min_val = pixel_type_t<image_view_type>::min();`
* `const auto max_val = pixel_type_t<image_view_type>::max();`
It's personal preference though...

Reply:
Sure, why not?